volume/nfs/nfs.go

```diff
@@ -71,26 +79,68 @@ func (v *Volume) DriverName() string {
 }
 
 func (v *Volume) Path() string {
-	return ""
+	v.m.Lock()
+	defer v.m.Unlock()
+	return v.hostFolder
 }
 
 func (v *Volume) Mount() (string, error) {
-	// The return value from this method will be passed to the container
-	return v.name, nil
-}
+	v.m.Lock()
+	defer v.m.Unlock()
 
-func (v *Volume) Unmount() error {
-	return nil
+	v.usedCount++
+	if v.usedCount > 1 {
+		// Already mounted
+		return v.hostFolder, nil
+	}
+	name, err := ioutil.TempDir(NFS_MOUNTS_FOLDER, "")
+	if err != nil {
+		return "", err
+	}
+	v.hostFolder = name
+	// retry=0,timeo=30: Fail if NFS server can't be reached in 30 second (no retries) - aggressive, but necessary because the Docker daemon becomes unresponsive if the mount command hangs.
+	args := []string{"-o", "retry=0,timeo=30"}
+
+	if err = libcontainer.DoMountCmd(v.DriverName(), v.Name(), v.hostFolder, args); err != nil {
+		return "", err
+	}
+	return v.hostFolder, nil
 }
 
-func (v *Volume) use() {
+func (v *Volume) Unmount() error {
 	v.m.Lock()
-	v.usedCount++
-	v.m.Unlock()
+	defer v.m.Unlock()
+
+	if err := v.release(); err != nil {
+		return err
+	}
+
+	// Don't unmount if still being used
+	if v.usedCount > 0 {
```

Comment:
You're never calling `release()`, so `usedCount` will never be 0. Did you test to see if the NFS volume actually gets unmounted when all containers that use it have been stopped?